test4-matching/src/main/java/Main.java

```diff
@@ -0,0 +1,110 @@
+import javafx.animation.PauseTransition;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.RowConstraints;
+import javafx.stage.Stage;
+import javafx.util.Duration;
+
+/** Class that runs the game */
+public class Main extends Application implements GameContext {
+
+    private ToggleButton[][] buttons;
+    private GameController gameController;
+    private PauseTransition pauseTransition;
+
+    /** launches the game */
+    public static void main(String[] args) {
+        Application.launch(args);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void start(Stage primaryStage) {
+        var parameters = getParameters().getRaw();
+        int n = Integer.parseInt(parameters.get(0));
```

Comment:
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0